nameable.rb

```diff
@@ -0,0 +1,28 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError, 'Subclasses must implement the correct_name method'
+  end
+end
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
```

Comment:
- Optional.
You can remove the `super()` as it is not necessary since the Nameable class does not have an explicit constructor.

Reply:
Hello @muneebulrehman I can not remove it because generate me a linter error but thanks for the explanation.